src/plugins/foxPage/hooks/useCurrentBlockNumber.ts

```diff
@@ -0,0 +1,33 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { getEthersProvider } from '../utils'
+
+export const useCurrentBlockNumber = () => {
+  const [block, setBlock] = useState<number | null>(null)
+  const blockNumberCallback = useCallback(
+    (blockNumber: number) => {
+      setBlock(blockNumber)
+    },
+    [setBlock],
+  )
```

Comment:
Stylistic, but does this need to be memoized? You can just use setBlock in the promise chain in the useEffect.

`    web3Provider
      .getBlockNumber()
      .then(setBlock)
      .catch(error => console.error(`Failed to get block number for chainId:`, error))`